api/src/main.py

```diff
@@ -193,12 +193,101 @@ async def get_token(
     )
 
 
-@app.get(
-    "/auth/me",
-    tags=["Auth"],
+@app.post(
+    "/dataset/validate",
+    tags=["Dataset"],
+    dependencies=[Depends(auth.get_username)],
+)
+def validate_dataset(dataset: schemas.Dataset):
+    # for all sample files in dataset, test access using raster /validate endpoint
+    # TODO this is commented out until the raster API fixes this endpoint
+    # https://github.com/NASA-IMPACT/delta-backend/issues/133
+
+    # for sample in dataset.sample_files:
+    #    url = f"{settings.raster_url}/cog/validate?url={sample}"
+    #    try:
+    #        response = requests.get(url)
+    #        if response.status_code != 200:
+    #            raise HTTPException(
+    #                status_code=response.status_code,
+    #                detail=(f"Unable to validate dataset: {response.text}"),
+    #            )
+    #    except Exception as e:
+    #        raise HTTPException(
+    #            status_code=422,
+    #            detail=(f"Sample file {sample} is invalid: {e}"),
+    #        )
+    return {
+        f"Dataset metadata is valid and ready to be published - {dataset.collection}"
+    }
+
+
+@app.post(
+    "/dataset/publish", tags=["Dataset"], dependencies=[Depends(auth.get_username)]
 )
+async def publish_dataset(dataset: schemas.Dataset):
+    # Construct and load collection
+    collection_data = {
+        "id": dataset.collection,
+        "title": dataset.title,
+        "description": dataset.description,
+        "license": dataset.license,
+        "extent": schemas.DatetimeExtent.parse_obj(
+            {
+                "spatial": {
+                    "bbox": [
+                        list(dataset.spatial_extent.dict(exclude_unset=True).values())
+                    ]
+                },
+                "temporal": {
+                    "interval": [
+                        # most of our data uses the Z suffix for UTC - isoformat() doesn't
+                        [
+                            x.isoformat().replace("+00:00", "Z")
+                            for x in list(
+                                dataset.temporal_extent.dict(
+                                    exclude_unset=True
+                                ).values()
+                            )
+                        ]
+                    ]
+                },
+            }
+        ),
+        "item_assets": {
```

Comment:
Spike for future work:
I think item assets should not be the responsibility of the data provider (or somehow configurable) rather than adding the item assets by default for all collections. When we ingest collections like the HLS data there will not be a default COG layer but we will want to provide an inventory of the assets provided in each item (see the assets in this item https://staging-stac.delta-backend.com/collections/hls-l30-002-ej-reprocessed/items/HLS.L30.T15RYP.2021294T163232.v2.0-ej-reprocessed)

Reply:
I thought about this more yesterday and am seeing how we might just just use some common components for a different type of collection publication path--this one checks all the boxes for a dashboard collection, a future alternate path might support custom types of collections. I'm going to resolve this conversation for now and we can develop on alternate data types in the future. Thanks for the :ack: